foundation/storage/src/main/kotlin/com/pingidentity/storage/DataStoreStorage.kt

```diff
@@ -41,10 +46,19 @@ class DataStoreStorage<T : @Serializable Any>(
     }
 
     /**
-     * Deletes the item from the DataStore.
+     * Deletes the item from the DataStore by setting it to null.
+     * This clears the data content but does not delete the physical file.
+     * To delete the physical file, use deleteFile() instead.
      */
     override suspend fun delete() {
         dataStore.updateData { null }
+        withContext(Dispatchers.IO) {
+            try {
+                file?.delete()
+            } catch (e: Exception) {
+                // Ignore file deletion errors
```

Comment:
Maybe we should at least log the errors. It will allow for customers debugging and surfacing the errors if any